flancy.psm1

```diff
@@ -156,6 +156,60 @@ function New-Flancy {
         [Parameter(Position=0)]
         [string] $url='http://localhost:8000',
         [Parameter(Mandatory=$false)]
+        [ValidateScript({
+            $PropSet = 'path', 'method', 'script'

```

Comment:
While my scriptblock works for now, it' really not that good, because if we try to extend it to support more options it will quickly became a mess.

Here is what I think we can do:

1. Define a dummy function inside the scriptblock and take advantage of the built-in validators:

        [ValidateScript({
            function Validate-WebSchema
            {
                [CmdletBinding()]
                Param
                (
                    [Parameter(Mandatory = $true, ValueFromPipelineByPropertyName = $true)]
                    [ValidateScript({
                        if($_.IsAbsoluteUri)
                        {
                            throw "'path' property can't be absolute URI: $($_.Path)"
                        }
                        $true
                    })]
                    [ValidateNotNullOrEmpty()]
                    [System.Uri]$Path,

                    [Parameter(Mandatory = $true, ValueFromPipelineByPropertyName = $true)]
                    [ValidateSet('Delete', 'Get', 'Head', 'Options', 'Post', 'Put', 'Patch')]
                    [string]$Method,

                    [Parameter(Mandatory = $true, ValueFromPipelineByPropertyName = $true)]
                    [ValidateNotNullOrEmpty()]
                    [scriptblock]$Script
                )

                Process {$true}
            }

            if($_.GetType().GetInterfaces().Name -notcontains 'IDictionary')
            {
				throw 'One of the the supplied objects is not a dictionary.'
			}

            $_ | ForEach-Object {
                try{ Validate-WebSchema @_ -ErrorAction Stop }
                catch { throw $_ }
            }
            $true
        })]
        [ValidateNotNullOrEmpty()]
        [object[]] $webschema = @(@{path='/';method='Get';script = {"Hello World!"}})

2. Change the way `New-Flancy` accepts parameters, so they can be validated separately:

        function New-Flancy
        {
            param(
                [System.Uri]$Url='http://localhost:8000',

                [Parameter(ValueFromPipelineByPropertyName = $true)]
                [ValidateScript({
                    if($_.IsAbsoluteUri)
                    {
                        throw "'path' property can't be absolute URI: $($_.Path)"
                    }
                    $true
                })]
                [ValidateNotNullOrEmpty()]
                [System.Uri]$Path = '/',

                [Parameter(Mandatory = $true, ValueFromPipelineByPropertyName = $true)]
                [ValidateSet('Delete', 'Get', 'Head', 'Options', 'Post', 'Put', 'Patch')]
                [string]$Method,

                [Parameter(Mandatory = $true, ValueFromPipelineByPropertyName = $true)]
                [ValidateNotNullOrEmpty()]
                [scriptblock]$Script,

                [switch] $Passthru,

                [switch] $Public
            )

            Begin {
                $xUrl = $null
                $WebSchema = @()
                $xPassthru = $null
                $xPublic = $null
            }

            Process {
                $psb = $PSBoundParameters
                $WebSchema += @{
                    Path = $Path
                    Method = $Method
                    Script = $Script
                }
                $xUrl = $Url
                $xPassthru = $Passthru
                $xPublic = $Public

            }

            End {
                if($WebSchema)
                {
                    $WebSchema
                    $xUrl
                    $xPassthru
                    $xPublic
                    # Do New-Flancy stuff here, using $WebSchema and $x* variables
                }

            }
        
   `New-Flancy` then could be invoked this way

        $WebSchema = @(
          @{Path = '/' ; Method = 'Get' ; Script = {"Hello World!"}},
          @{Path = '/api' ; Method = 'Get' ; Script = {"Hello Api!"}}
        )

        $WebSchema | ForEach-Object {
            New-Object psobject -Property $_
         } | New-Flancy -Url ''http://localhost:5555' -Public -Passthru

What do you think?